### PR TITLE
(maint) detail v3->v4 API changes

### PR DESCRIPTION
**Pull request by wkalt**
This documents the changes we've made from the v3->v4 API.

### CODE REVIEW

**Comment by wkalt** [comment]
this is a slight rewording of the confluence doc here: https://confluence.puppetlabs.com/display/PP/Upcoming+PuppetDB+v4+API+changes

I'm anticipating we may be pointing some people at it before release time, so I've worded it that way to avoid confusion. We'll need to update it at release time to make sure tense is appropriate and the "upcoming" section has been updated. For that I created this ticket https://tickets.puppetlabs.com/browse/PDB-1281

File: documentation/api/query/v4/upgrading-from-v3.markdown [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+* The `in` and `extract` operators have been changed to accept multiple fields,
+  allowing more concise subquerying as explained here:
+  https://github.com/puppetlabs/puppetdb/pull/1053
+
+### CHANGES STILL TO COME
+
+  * add endpoints for dedicated querying of report logs and metrics
+  * add `state-overview` endpoint for aggregate count queries over last report
+  statuses (Puppet Enterprise only.)
```
Comment (wkalt) [inline documentation/api/query/v4/upgrading-from-v3.markdown]: @senior @turbodog @LaMetterey @kbarber do we require any delicacy here?

**Comment by wkalt** [comment]
this will also require a puppet-docs update when it goes in

**Comment by rbrw** [comment]
@pljenkinsro retest this please

**Pull request closed without merging**